### PR TITLE
Add TaxCountryRegion & TaxCode codes to PT regime

### PR DESCRIPTION
**Pull request by cavalle**
* Adds `TaxCountryRegion` and `TaxCode` codes to the PT regime
* Adds table documentation on how to map GOBL fields to those codes
* Enables replacing hard-coded mapping logic in IX provider (see https://github.com/invopop/invoicexpress/pull/35)

### CODE REVIEW

File: regimes/pt/README.md [hunk 1]
```diff
@@ -21,9 +21,43 @@ Portugal doesn't have an e-invoicing format per se. Tax information is reported
 | ND | Nota de débito | `credit-note` | |
 | NC | Nota de crédito | `debit-note` | |
 
+### `TaxCountryRegion` (País ou região do imposto)
+
```
Comment (samlown) [inline regimes/pt/README.md:25]: I think we need more details here for how these fields will be used, I tried to do the same for Italy already. The "GOBL Tax Identity Zone" also feels a bit out of place, it looks like we made up a number. These are actually the zone codes from ISO 3166-2:PT (according to the comment in the code: https://github.com/invopop/gobl/blob/main/regimes/pt/pt.go#L23)

File: regimes/pt/tax_categories.go [hunk 2]
```diff
@@ -135,6 +144,7 @@ var taxCategories = []*tax.Category{
 					i18n.PT: "Artigo 16.°, n.° 6 do CIVA",
 				},
 				Codes: cbc.CodeSet{
+					KeyATTaxCode:          "ISE",
```
Comment (samlown) [inline regimes/pt/tax_categories.go:147]: I'm surprised the linter didn't pick up on this... might be better to have constants when the code is repeated so much?

File: regimes/pt/README.md (hunk 1 above)
Comment (cavalle) [reply regimes/pt/README.md:27]: Local codes descriptions added in f055a5bde1a0e2f7dcdff3ab3ffaea1bd95e54af. Thank you for the feedback!

File: regimes/pt/tax_categories.go (hunk 2 above)
Comment (cavalle) [reply regimes/pt/tax_categories.go]: Repeated codes extracted to constants in 82b32dbb5066a08dc8417aaf22532d6edbe4608e. Thank you for the feedback!

**Pull request merged**